File: src/main/ui/src/models/core/PDOperationMeta.ts
import type {PDMemberMeta} from "@/models/core/PDMemberMeta";
import type {PDAttributeMeta} from "@/models/core/PDAttributeMeta";

export interface PDOperationMeta extends PDMemberMeta {
    icon?: string;
}


export function isPDOperationMeta(obj: any): obj is PDOperationMeta {
    return 'icon' in obj;
}

File: src/main/java/de/fhswf/genericapplication/models/core/PDOperationMeta.java
package de.fhswf.genericapplication.models.core;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlAttribute;
import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement(namespace = "core", name = "operation")
@XmlAccessorType(XmlAccessType.FIELD)
public class PDOperationMeta extends PDMemberMeta {
    @XmlAttribute
    @JsonProperty
    private String icon;
}
